xsnippet/api/resource.py

```diff
@@ -58,7 +61,37 @@ def __init__(self, *args, **kwargs):
         #: an application database alias to make code a bit readable
         self.db = self.request.app['db']
 
-    def make_response(self, data, headers=None, status=200):
+    @asyncio.coroutine
+    def __iter__(self):
+        # So far (Jan 5, 2018) aiohttp doesn't support custom request classes,
+        # but we would like to provide a better user experience for consumers
+        # of this middleware. Hence, we monkey patch the instance and add a new
+        # async method that would de-serialize income payload according to HTTP
+        # content negotiation rules.
+        setattr(self.request.__class__, 'get_data', self._get_data())
```

Comment:
I know that some people would not approve, but I do not think we should follow all OOP rules :P This seems to be a simpler solution.

Reply:
It's just for start. I think we can come up with proxy object once it's a standalone solution (in separate package) but for the sake of simplicity now I decided to stick with monkey-patching. :)